etc/schema.json

```diff
@@ -1712,54 +1712,68 @@
             "additionalProperties": false,
             "properties": {
                 "end": {
-                    "type": "integer"
+                    "type": "integer",
+                    "description": "The offset of the last byte captured"
                 },
                 "file_id": {
-                    "type": "integer"
+                    "type": "integer",
+                    "description": "Integer value representing the id of a file that has been stored"
                 },
                 "filename": {
-                    "type": "string"
+                    "type": "string",
+                    "description": "Name of the file as observed in network traffic"
                 },
                 "gaps": {
-                    "type": "boolean"
+                    "type": "boolean",
+                    "description": "Boolean value indicating if there were gaps in the file"
                 },
                 "magic": {
-                    "type": "string"
+                    "type": "string",
+                    "description": "[optional, requires libmagic] The magic value for the file"
                 },
                 "md5": {
-                    "type": "string"
+                    "type": "string",
+                    "description": "[optional, if state is ``CLOSED``] When closed, md5 sum"
                 },
                 "sha1": {
-                    "type": "string"
+                    "type": "string",
+                    "description": "[optional, if state is ``CLOSED]`` When closed, sha1 sum"
```

Comment:
There are several optional fields like that, are we starting a new scheme to add that in their description?

Reply:
I'd add this optional annotation to the end, it will be more readable in the generated documentation that way (https://docs.suricata.io/en/latest/appendix/eve-index.html).

Reply:
>I'd add this optional annotation to the end, it will be more readable in the generated documentation that way (https://docs.suricata.io/en/latest/appendix/eve-index.html).

@jasonish 
This will involve changing the output format - is that ok for a patch release?

Reply:
> > I'd add this optional annotation to the end, it will be more readable in the generated documentation that way (https://docs.suricata.io/en/latest/appendix/eve-index.html).
> 
> @jasonish This will involve changing the output format - is that ok for a patch release?

What changes exactly? Currently there is no description in the appendix, so adding one is good.

Reply:
I feel like this thread looped over itself...

I think Jeff's PR here is good